### ISSUE TITLE
Summary of donors, specimens and cells are 0 for `DCP/2 Analysis`

### ISSUE THREAD

**Issue by theathorn** [DataBiosphere/azul#3281]
On the "Projects" tab select "DCP/2 Analysis" from the "File" -> "File Source" dropdown.
The summary counts for "Donors", "Specimens" and "Estimated Cells" all show as zero.

Daniel and Noah confirmed that the /index/summary endpoint is returning zero values for donorCount, specimenCount and totalCellCount while /index/projects/hits has many non-zero values for these entities.

![image.png](https://images.zenhubusercontent.com/5ae772614b5806bc2bd13821/6a13b2e1-5899-41eb-b8d8-9def0838b8cd)

**Comment by theathorn** [comment]
Spike to investigate root cause.

**Comment by noah-aviel-dove** [comment]
Analysis processes can only result in file-type outputs, so in order to match the filter a hit must be either:
1. An outer `files` entity
2. An entity with inner `files` entities, e.g., a project.

Each summary field is calculated by querying a specific index (e.g. `projectCount` comes from `projects`), and the indices are scanned independently. The "missing" fields are derived from the `specimens` index (`donorCount`, `specimenCount`) and the `cell_suspensions` index (`totalCellCount`).

Samples/specimens do not have inner file entities, only a summary of files types (note that filtering by file source is not an option on the "Samples" tab in the browser). As such, there are no hits when querying the `specimens` index using the `fileSource` filter. This is reflected in the `/index/samples` endpoint (no hits) and the "Samples" tab on the browser (no results). For the same reason, the summary fields that are derived from the `specimens` index, including donor and specimen count, are empty as well. 

To summarize (no pun intended), the browser results appear contradictory because the summary is generated from multiple queries over heterogeneous entity types, and some entity types cannot pass the filter due to missing inner entities.

**Comment by melainalegaspi** [comment]
@hannes-ucsc to review Noah's findings.

**Comment by hannes-ucsc** [comment]
> Samples/specimens do not have inner file entities, only a summary of files types (note that filtering by file source is not an option on the "Samples" tab in the browser). As such, there are no hits when querying the specimens index using the fileSource filter.

Don't completely agree with the above diagnosis. Inner entities of type `I` are always summaries in an outer entity of type `O` except when `I == O` or `I == "files" and O == "bundles"`. And whether inner entities are grouped or not doesn't affect the filters. The grouping adds detail but loses none.

I think the issue is that the indexer produces no contributions to `samples` or `cell_suspension` entities from stitched-on primary subgraphs when it indexes an analysis subgraph. Their suppression was added as part of the fix for #2868 but the negative consequences weren't completely clear to me at the time.

With respect to the zero donor and specimen counts, this is a duplicate of #3905. The zero cell count has a similar cause so we'll leave this issue open to track that.

In essence, this change should fix the zero summary cell count:

```python
Index: src/azul/plugins/metadata/hca/transform.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/plugins/metadata/hca/transform.py b/src/azul/plugins/metadata/hca/transform.py
--- a/src/azul/plugins/metadata/hca/transform.py	(revision 3968bc5c6cd29142dbaa62a0f5d23fc90222d559)
+++ b/src/azul/plugins/metadata/hca/transform.py	(date 1629412847068)
@@ -1332,7 +1332,7 @@
 
     def transform(self) -> Iterable[Contribution]:
         project = self._get_project(self.api_bundle)
-        for cell_suspension in api.not_stitched(self.api_bundle.biomaterials.values()):
+        for cell_suspension in self.api_bundle.biomaterials.values():
             if isinstance(cell_suspension, api.CellSuspension):
                 samples: MutableMapping[str, Sample] = dict()
                 self._find_ancestor_samples(cell_suspension, samples)
```

**Comment by hannes-ucsc** [comment]
Consider solving this and #3095 in the same PR.

**Comment by hannes-ucsc** [comment]
For demo, attempt to reproduce in prod.
